tests/tests_user_views.py

```diff
@@ -0,0 +1,124 @@
+PASSWORD = "Aa313611"
+EMAIL = "testuser@gmail.com"
+USER_NAME_NEW = "tests_user_new"
+PHONE_NUMBER = "0502220000"
+CITY = "Test City"
+
+
+@pytest.fixture
+def new_profile():
+    return Profile.create(username=USERNAME, password=PASSWORD, account_type=Account_type.STUDENT,
+                          first_name=FIRSTNAME, last_name=LASTNAME)
+
+
+@pytest.fixture
+def client():
+    return Client()
+
+
+@pytest.fixture
+def authenticated_client(client, new_user):
+    new_user.save()
+    client.login(username=USERNAME, password=PASSWORD)
+    return client
+
+
+@pytest.fixture
+def valid_registration_data():
+    return {
+        'username': 'test_user',
+        'email': 'test_user@test.com',
+        'password1': 'test_password',
+        'password2': 'test_password',
+        'account_type': Account_type.TEACHER,
+        'phone_number': '+1234567890',
+        'city': 'Test City',
+        'meeting_method': Meeting_method.LIVE
+    }
+
+
+@pytest.mark.django_db
+class TestProfileViews:
+    def test_users_detail_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        users_detail_response = authorized_client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'profile_details.html' in users_detail_response.templates[0].name
+
+    def test_users_detail_view_loaded_unauthorized(self, client, persist_user_profile):
+        users_detail_response = client.get(f'/users/{persist_user_profile.user.username}/', follow=True)
+        assert users_detail_response.status_code == 200
+        assert 'main/login.html' in users_detail_response.templates[0].name
+
+    def test_edit_profile_view_unauthenticated(self, client, new_profile):
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_edit_profile_view_authenticated(self, client, new_profile):
+        client.login(username='testuser', password='testpassword')
+        url = reverse('users-edit-profile', kwargs={'slug': new_profile.user.username})
+        response = client.get(url, follow=True)
+        assert response.status_code == 200
+        assert 'main/login.html' in response.templates[0].name
+
+    def test_users_update_view_loaded_authorized(self, authorized_client, persist_user_profile):
+        user_detail_response = authorized_client.get(f"/users/{persist_user_profile.pk}/edit-profile/")
+        assert user_detail_response.status_code == 200
+        assert 'edit-profile.html' in user_detail_response.templates[0].name
+
+    @pytest.mark.parametrize("email, city", [(EMAIL, "city_test")])
+    def test_update_user(self, persist_user_profile, authorized_client, email, city):
+
+        profile = Profile.objects.filter(user=persist_user_profile.user).first()
```

Comment:
Please use a fixture for this

Reply:
There is a problem with the persist_user_profile fixture, it doesn't let you update fields and that's the only way it worked